src/ecalc/libraries/libecalc/common/libecalc/core/consumers/compressor/component.py

```diff
@@ -90,20 +90,25 @@ def evaluate(
             else:
                 aggregated_result.extend(model_result)
 
+        energy_usage = TimeSeriesRate(
+            values=aggregated_result.energy_usage,
+            timesteps=evaluated_timesteps,
+            unit=aggregated_result.energy_usage_unit,
+            regularity=regularity,
+        )
+
+        if energy_usage.unit == Unit.STANDARD_CUBIC_METER_PER_DAY:
+            energy_usage = energy_usage.to_calendar_day()  # legacy consumer converts to calendar day if fuel consumer
+
         component_result = core_results.CompressorResult(
             timesteps=evaluated_timesteps,
             power=TimeSeriesRate(
                 values=aggregated_result.power,
                 timesteps=evaluated_timesteps,
                 unit=aggregated_result.power_unit,
                 regularity=regularity,
-            ),
-            energy_usage=TimeSeriesRate(
-                values=aggregated_result.energy_usage,
-                timesteps=evaluated_timesteps,
-                unit=aggregated_result.energy_usage_unit,
-                regularity=regularity,
-            ),
+            ).fill_nan(0.0),
```

Comment:
Fill nan because this is done in legacy. Should we handle this some other way?

Reply:
Think we should stick to v1 for comparison reasone initially, yes. Then make an overall decision across "everything" on how we should handle NaN, instead of deciding on a per case basis...